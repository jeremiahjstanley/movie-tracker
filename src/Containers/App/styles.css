.app {
  display: grid;
  grid-template-columns: 40% 60%;
  letter-spacing: .1em;
  text-align: center;
}

.app-header {
  background-color: #111;
  display: flex;
  grid-column: 1/3;
  height: 3rem;
  justify-content: space-between;
  position: sticky;
}

.favorite {
  box-shadow: inset 0 0 1em .5em #B9090B;
}

button {
  background-color: #B9090B;
  border: 0;
  color: #F3F3F3;
  font-size: 1em;
  font-weight: 100;
  padding: .5em;
}

