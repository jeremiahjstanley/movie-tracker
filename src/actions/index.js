export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies
});

export const toggleFavorite = (id) => ({
  type: 'TOGGLE_FAVORITE',
  id
});

export const signUp = (username, password, email) => ({
  type: 'SIGN_UP',
  username,
  password, 
  email
});

export const logIn = (username, password) => ({
  type: 'LOG_IN',
  username,
  password
});

export const logOut = (username) => ({
  type: 'LOG_OUT',
  username
});

