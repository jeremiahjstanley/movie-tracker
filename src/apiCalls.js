
const fetchMovieData = async (key) => {
  const url = `https://api.themoviedb.org/3/movie/550?api_key=${key}`;
  const response = await fetch(url);
  const results = await response.json();
  return results;
};

export default fetchMovieData;