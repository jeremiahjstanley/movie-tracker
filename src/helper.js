export const movieCleaner = results => {
  return {
    title: results.original_title,
    releaseDate: results.release_date,
    rating: results.vote_average,
    tagline: results.tagline,
    movieImage: results.poster_path,
    overview: results.overview,
    favorite: false
  };
};