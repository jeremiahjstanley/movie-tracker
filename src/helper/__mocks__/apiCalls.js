export const fetchMovieData = jest.fn();

export const deleteFavoriteFromDatabase = jest.fn();

export const getFavoritesFromDatabase = jest.fn().mockImplementation(() => ({
  data: [{title: 'Con Air'}, {title: 'Face/Off'}, {title: 'Captain Ron'}]
}))

export const createUser = jest.fn().mockImplementation(() => ({
  id: 2
}));

export const sendFavoriteToDatabase = jest.fn()
