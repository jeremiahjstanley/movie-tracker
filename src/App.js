import React, { Component } from 'react';
import './App.css';
import fetchMovieData from './apiCalls';
import { apiKey } from './apiKey';
import { movieCleaner } from './helper';
import { connect } from 'react-redux';
import { addMovies } from './actions';

class App extends Component {

  componentDidMount = async () => {
    const movies = await fetchMovieData(apiKey);
    const cleanMovies = await movieCleaner(movies);
    this.props.addMovies(cleanMovies);
  }

  render() {
    return (
      <div className="App">
        hello world
      </div>
    );
  }
}
export const mapStateToProps = (state) => ({
  movies: state.movies
});

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(addMovies(movies))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
