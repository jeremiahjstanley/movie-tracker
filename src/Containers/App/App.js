import React, { Component } from 'react';
import './App.css';
import CardContainer from '../Containers/CardContainer';
import MovieDetails from '../MovieDetails';
import { connect } from 'react-redux';
import { NavLink, Route } from 'react-router-dom';

class App extends Component {

  render() {
    console.log()
    return (
      <div className="App">

          Hello World
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  movies: state.movies
});



export default connect(mapStateToProps)(App);
