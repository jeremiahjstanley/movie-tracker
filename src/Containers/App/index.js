import React, { Component } from 'react';
import CardContainer from '../CardContainer';
import './App.css';
import { NavLink, Route, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import MovieDetails from '../MovieDetails';
import LoginForm from '../LoginForm';
import SignUpForm from '../SignUpForm';


class App extends Component {

  
  render() {
    return (
      <div className="App">

        <header className="app-header">
          <NavLink to='/login'>
            Login/LogOut
          </NavLink>
          <NavLink to='/signup'>
            SignUp
          </NavLink>
          <NavLink to='/favorites'>
            favorites
          </NavLink>
        </header>

        <Route path='/movies/:title' render={({match}) => {
          const movieToDisplay=this.props.movies.find(movie => movie.title === match.params.title)
          return <MovieDetails {...movieToDisplay}/>
        }}/>
        <Route path='/login' component={LoginForm}/>
        <Route path='/signup' component={SignUpForm}/>
        <Route path='/' component={CardContainer}/>
        
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  movies: state.movies
});


export default withRouter(connect(mapStateToProps)(App));
