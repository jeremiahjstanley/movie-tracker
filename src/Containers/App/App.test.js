import React from 'react';
import { mount, shallow } from 'enzyme';
import { logIn, logOut, addFavorite, updateFavorites, getUserFavorites } from '../../actions';
import { App, mapStateToProps, mapDispatchToProps } from '../App';
import { getFavoritesFromDatabase, sendFavoriteToDatabase, deleteFavoriteFromDatabase } from '../../helper/apiCalls';
import { BrowserRouter } from 'react-router-dom';

jest.mock('../../helper/apiCalls');

describe('App tests', () => {
    let wrapper; 
    let mockFavorites;
    let mockMovies;
    let mockUser;
    let mockUpdateFavorites;
    let mockAddToFavorites;
    let mockHistory;
    let mockGetUserFavorites;
    let mockLogOutUser;
    let mockLogInUser;
    let mockLogIn;

  beforeEach(() => {
    mockMovies = [{ title: 'ConAir', id: 7, poster_path: 'google.com' }, { title: 'FaceOff', id: 8, poster_path: 'bing.com' }];
    mockFavorites = [{ title: 'ConAir', movie_id: 7, id:7 }];
    mockUser = {email: 'nick@cage.com', name: 'Nick', id:2};
    mockUpdateFavorites = jest.fn();
    mockAddToFavorites = jest.fn();
    mockHistory = [];
    mockGetUserFavorites = jest.fn();
    mockLogOutUser = jest.fn();
    mockLogInUser = jest.fn();
    mockLogIn = jest.fn()

    wrapper = shallow(
      <App 
        favorites={mockFavorites}
        movies={mockMovies}
        users={mockUser}
        updateFavorites={mockUpdateFavorites}
        addToFavorites={mockAddToFavorites}
        history={mockHistory}
        getUserFavorites={mockGetUserFavorites}
        logOutUser={mockLogOutUser}
        logInUser={mockLogInUser}
        logIn={mockLogIn}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should invoke getUser method on page load', async () => {
    const spy = spyOn(wrapper.instance(), 'getUser');
    
    wrapper.unmount()
    // wrapper.shallow()
    // wrapper.mount()
    // wrapper.instance().getUser()

    await expect(spy).toHaveBeenCalled()
  });

  it.skip('should log in a user, if present in localStorage', async () => {
    await localStorage.setItem('user', JSON.stringify(mockUser));

    await wrapper.instance().getUser();

    await expect(mockLogInUser).toHaveBeenCalledWith(mockUser.email, mockUser.name, mockUser.id)
  })

  it.skip('return a users favorites from the database', async () => {
    await localStorage.setItem('user', JSON.stringify(mockUser));

    await wrapper.instance().getUser();

    await expect(getFavoritesFromDatabase).toHaveBeenCalledWith(mockUser.id)
  })

  it.skip('return update the users favorites', async () => {
    await localStorage.setItem('user', JSON.stringify(mockUser));

    await wrapper.instance().getUser();

    await expect(mockUpdateFavorites).toHaveBeenCalled()
  })

  it.skip('should check to see if the user has favorites', () => {
    wrapper.instance().checkFavorites(7)

    const spy = spyOn(wrapper.instance(), 'findMovie');

    expect(spy).toHaveBeenCalledWith(7)
  })

  it.skip('should add a movie to the users favorites', () => {
    wrapper.instance().checkFavorites(8)

    const spy = spyOn(wrapper.instance(), 'addFavorite');
    
    expect(spy).toHaveBeenCalled()
  })

  it.skip('should remove a movie from the users favorites', () => {
    wrapper.instance().checkFavorites(7)

    const spy = spyOn(wrapper.instance(), 'removeFavorite');
    
    expect(spy).toHaveBeenCalled()
  })

  it('should find a movie', () => {
    const results = wrapper.instance().findMovie(7)

    const expected = {id: 7, poster_path: 'google.com', title: 'ConAir'}
    
    expect(results).toEqual(expected)
  })

  it('should check for a favorite movie', () => {
    const expected = {favorite: true, id: 9, poster_path: 'bing.com', title: 'Face/Off'}
    
    wrapper.instance().checkFavorites(8)

    expect(mockAddToFavorites).toHaveBeenCalledWith(mockMovies[1])
  })

  it('should add a movie to the favorites array', () => {
    const expected = {favorite: true, id: 8, poster_path: 'bing.com', title: 'Face/Off'}
    
    wrapper.instance().addFavorite(expected)

    expect(mockAddToFavorites).toHaveBeenCalledWith(expected)
  })

  it('should store the users favorites in the database', () => {
    const expected = {favorite: true, id: 8, poster_path: 'bing.com', title: 'Face/Off'}
    
    wrapper.instance().addFavorite(expected)

    expect(sendFavoriteToDatabase).toHaveBeenCalledWith(expected, mockUser.id)
  })

  it('should promp the user to login attempt to add a favorited without being signed in', () => {
    mockUser = {};
    wrapper = shallow(
      <App 
        favorites={mockFavorites}
        movies={mockMovies}
        users={mockUser}
        updateFavorites={mockUpdateFavorites}
        addToFavorites={mockAddToFavorites}
        history={mockHistory}
      />);
    
    wrapper.instance().checkFavorites(8);

    expect(mockHistory).toEqual(['/login']);
  })

  it('should remove a movie from the favorites array', () => {
    const expected = {favorite: true, id: 8, poster_path: 'bing.com', title: 'Face/Off'}
    
    wrapper.instance().removeFavorite(expected);

    expect(mockUpdateFavorites).toHaveBeenCalledWith(mockFavorites);
  })

  it('should deleted a unfavorited movie from the database', () => {
    const expected = {favorite: true, id: 8, poster_path: 'bing.com', title: 'Face/Off'}
    
    wrapper.instance().addFavorite(expected);

    expect(deleteFavoriteFromDatabase).toHaveBeenCalledWith(expected.id, mockUser.id)
  })

  it('should call getUserFavorites with an empty array ', () => {
    wrapper.instance().logOut();

    expect(mockGetUserFavorites).toHaveBeenCalledWith([]);
  })

  it('should call logOutUser', () => {
    wrapper.instance().logOut()

    expect(mockLogOutUser).toHaveBeenCalled();
  })

  it('should redirect the user to the home screen', () => {
    wrapper.instance().logOut()

    expect(mockHistory).toEqual(['/']);
  })

  describe('mapStateToProps', () => {
    it('should return an object with user object and favorites array', () => {
      const mockState = { login: { email: 'nickcage@aol.com', name: 'Nick Cage', id: 1}, movies: [], favorites: []};
      const expected = {favorites: [], movies: [], users: {email: 'nickcage@aol.com', id: 1, name: 'Nick Cage'}}
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch when the form is submitted to log the user in', () => { 
      const mockGetUserFavorites = jest.fn();
      const mockLogInUser = jest.fn();

      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch)
      const actionToDispatch = logIn('nickcage@aol.com', 'Nick', 1)
      mappedProps.logInUser('nickcage@aol.com', 'Nick', 1)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch when the user clicks the logOut link', () => { 
      const mockGetUserFavorites = jest.fn();
      const mockLogInUser = jest.fn();

      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch)
      const actionToDispatch = logOut()
      mappedProps.logOutUser()

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch when adds a favorite', () => { 
      const mockGetUserFavorites = jest.fn();
      const mockLogInUser = jest.fn();

      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch)
      const actionToDispatch = addFavorite({title: 'Con Air'})
      mappedProps.addToFavorites({title: 'Con Air'})

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch when the form is submitted to retrieve the users favorites', () => { 
      const mockGetUserFavorites = jest.fn();
      const mockLogInUser = jest.fn();

      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch)
      const actionToDispatch = updateFavorites(['Con Air'])
      mappedProps.getUserFavorites(['Con Air'])

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch when a movie is favorited/unfavorited to update the favorites in the store', () => { 
      const mockGetUserFavorites = jest.fn();
      const mockLogInUser = jest.fn();

      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch)
      const actionToDispatch = updateFavorites(['Con Air'])
      mappedProps.updateFavorites(['Con Air'])

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  });
});