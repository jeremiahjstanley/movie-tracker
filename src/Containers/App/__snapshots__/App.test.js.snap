// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addToFavorites={[Function]}
    favorites={
        Array [
            Object {
              "id": 7,
              "movie_id": 7,
              "title": "ConAir",
            },
          ]
    }
    getUserFavorites={[Function]}
    history={Array []}
    logInUser={[Function]}
    logOutUser={[Function]}
    movies={
        Array [
            Object {
              "id": 7,
              "poster_path": "google.com",
              "title": "ConAir",
            },
            Object {
              "id": 8,
              "poster_path": "bing.com",
              "title": "FaceOff",
            },
          ]
    }
    updateFavorites={[Function]}
    users={
        Object {
            "email": "nick@cage.com",
            "id": 2,
            "name": "Nick",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Connect(Header))
          logOut={[Function]}
/>,
        <Route
          path="/favorites/"
          render={[Function]}
/>,
        <Route
          path="/movies/:title"
          render={[Function]}
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/login"
/>,
        <Route
          component={[Function]}
          path="/signup"
/>,
        <Route
          path="/"
          render={[Function]}
/>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "logOut": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/favorites/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/movies/:title",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/signup",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Connect(Header))
            logOut={[Function]}
/>,
          <Route
            path="/favorites/"
            render={[Function]}
/>,
          <Route
            path="/movies/:title"
            render={[Function]}
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/login"
/>,
          <Route
            component={[Function]}
            path="/signup"
/>,
          <Route
            path="/"
            render={[Function]}
/>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "logOut": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/favorites/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/movies/:title",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/signup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
